include/boost/beast/websocket/detail/mask.hpp

```diff
@@ -83,181 +84,59 @@ using maskgen = maskgen_t<std::minstd_rand>;
-        auto const i = reinterpret_cast<
-            std::uintptr_t>(p) & (sizeof(key)-1);
-        switch(i)
-        {
-        case 1: p[2] ^= static_cast<std::uint8_t>(key >> 16); BOOST_FALLTHROUGH;
-        case 2: p[1] ^= static_cast<std::uint8_t>(key >> 8);  BOOST_FALLTHROUGH;
-        case 3: p[0] ^= static_cast<std::uint8_t>(key);
-        {
-            auto const d = static_cast<unsigned>(sizeof(key) - i);
-            key = ror(key, 8*d);
-            n -= d;
-            p += d;
-            BOOST_FALLTHROUGH;
-        }
-        default:
-            break;
-        }
-    }
-
-    // Mask 4 bytes at a time
-    for(auto i = n / sizeof(key); i; --i)
-    {
-        *reinterpret_cast<
-            std::uint32_t*>(p) ^= key;
-        p += sizeof(key);
-    }
-
-    // Leftovers
-    n &= sizeof(key)-1;
-    switch(n)
-    {
-    case 3: p[2] ^= static_cast<std::uint8_t>(key >> 16); BOOST_FALLTHROUGH;
-    case 2: p[1] ^= static_cast<std::uint8_t>(key >> 8);  BOOST_FALLTHROUGH;
-    case 1: p[0] ^= static_cast<std::uint8_t>(key);
-        key = ror(key, static_cast<unsigned>(8*n));
-        BOOST_FALLTHROUGH;
-    default:
-        break;
-    }
-}
-
-// 64-bit optimized
+// Apply mask in place
 //
-template<class = void>
+inline
 void
-mask_inplace_fast(
-    boost::asio::mutable_buffer const& b,
-        std::uint64_t& key)
+mask_inplace(boost::asio::mutable_buffer& b, prepared_key& key)
 {
     auto n = b.size();
-    auto p = reinterpret_cast<std::uint8_t*>(b.data());
-    if(n >= sizeof(key))
+    auto mask = key; // avoid aliasing
+    auto p = reinterpret_cast<unsigned char*>(b.data());
+    while(n >= 4)
     {
-        // Bring p to 8-byte alignment
-        auto const i = reinterpret_cast<
-            std::uintptr_t>(p) & (sizeof(key)-1);
-        switch(i)
-        {
-        case 1: p[6] ^= static_cast<std::uint8_t>(key >> 48); BOOST_FALLTHROUGH;
-        case 2: p[5] ^= static_cast<std::uint8_t>(key >> 40); BOOST_FALLTHROUGH;
-        case 3: p[4] ^= static_cast<std::uint8_t>(key >> 32); BOOST_FALLTHROUGH;
-        case 4: p[3] ^= static_cast<std::uint8_t>(key >> 24); BOOST_FALLTHROUGH;
-        case 5: p[2] ^= static_cast<std::uint8_t>(key >> 16); BOOST_FALLTHROUGH;
-        case 6: p[1] ^= static_cast<std::uint8_t>(key >> 8);  BOOST_FALLTHROUGH;
-        case 7: p[0] ^= static_cast<std::uint8_t>(key);
-        {
-            auto const d = static_cast<
-                unsigned>(sizeof(key) - i);
-            key = ror(key, 8*d);
-            n -= d;
-            p += d;
-            BOOST_FALLTHROUGH;
-        }
-        default:
-            break;
-        }
+        for(int i = 0; i < 4; ++i)
+            p[i] ^= mask[i];
+        p += 4;
+        n -= 4;
     }
-
-    // Mask 8 bytes at a time
-    for(auto i = n / sizeof(key); i; --i)
+    if(n > 0)
     {
-        *reinterpret_cast<
-            std::uint64_t*>(p) ^= key;
-        p += sizeof(key);
+        rol(key, 4 - n);
```

Comment:
I know this is inconsequential because `mask` is used in the loop, but doing the `rol` before the loop trips up the readers eye. Consider doing the `rol` after the loop.